Dockerfile.osbase

```diff
@@ -1,4 +1,8 @@
 FROM quay.io/centos/centos:stream9
-RUN dnf -y install python3-pip rust cargo && ln -s /usr/bin/python3.9 /usr/bin/python
+RUN dnf -y install python3.12 python3.12-pip rust cargo \
+ && ln -s /usr/bin/python3.12 /usr/bin/python \
+ && rm -f /usr/bin/python3 \
+ && ln -s /usr/bin/python3.12 /usr/bin/python3
```

Comment:
If you're going to do this, consider using `ln -sf` rather than `rm` (and, I would give the same treatment to both `/usr/bin/python` and `/usr/bin/python3`); however, I think that there are less hacky ways to accomplish this ([e.g.](https://linux.die.net/man/8/alternatives)).